status_dashboard.py

```diff
@@ -0,0 +1,97 @@
+from retriever import install_sqlite
+from retriever.lib.engine_tools import getmd5
+from retriever import datasets
+from retriever.engines import engine_list
+from tempfile import mkdtemp
+from shutil import rmtree
+import os
+import urllib.request
+import json
+from datetime import datetime
+
+mysql_engine, postgres_engine, sqlite_engine, msaccess_engine, csv_engine, download_engine, json_engine, xml_engine = engine_list
+
+file_location = os.path.dirname(os.path.realpath(__file__))
+
+example_datasets = ['bird-size', 'mammal-masses', 'airports', 'portal']
+precalculated_md5 = {'airports': 'f16a0bb21d347e2ba180cde26f35c87a',
+                     'bird-size': '98dcfdca19d729c90ee1c6db5221b775',
+                     'mammal-masses': '6fec0fc63007a4040d9bbc5cfcd9953e',
+                     'portal': 'aaa2b66a3a77efe416e3555c444f4e2e'}
+
+
+def check_urls_exist(dataset):
+    """Check to see if all resources of a dataset
+     exist at given urls and return a dictionary of failing ones."""
+    failed = {}
+    for (url_key, url) in dataset.tables.items():
+        request = urllib.request.Request(url)
+        request.get_method = lambda: 'HEAD'
+        try:
+            urllib.request.urlopen(request)
+        except urllib.request.HTTPError as e:
+            failed[url_key] = (url, e.code)
+    return failed
+
+
+def get_last_modified(dataset):
+    """Returns a dictionary of tables with the last modified date as
```

Comment:
```
Returns a dictionary of tables with the last modified date as
 values for particular dataset and None if it cannot find it.
```

I don't understand what you are trying to say

Reply:
It returns a dictionary of particular resources of a dataset with their last modified dates.
For eg. consider airports dataset then it will return something like
```python 
{'regions':python datetime,'navaids':python datetime ...... ,....... ,.....}
```

Reply:
@henrykironde I will update the comments so it becomes clearer what each of the functions are doing.